### PR TITLE
Adds a filter to segment lapsing trialist

### PR DESCRIPTION
**Pull request by GlynnPhillips**
The value is the number of days before their trial ends that we want to
segment them.

```
{
    type: daysLeftOnTrial,
    value: 10
}
```

@tom-parker @commuterjoy

### CODE REVIEW

File: server/filters/lapsingtrialists.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+	.then(res => {
+		if (res.status !== 200) {
+			return false;
+		} else {
+			return res.json().then(subs => {
+				if(!subs.b2c.triallist) {
+					return false;
+				}
+
+				const trialSub = subs.b2c.subscriptions.filter((subscription) => {
```
Comment (commuterjoy) [inline server/filters/lapsingtrialists.js]: perhaps check `subscriptions` exists?

File: server/filters/lapsingtrialists.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+if (!process.env.USER_STATUS_API_KEY) {
+	throw new Error('You need to set USER_STATUS_API_KEY in your environment');
+}
+
+const metrics = require('@financial-times/n-express').metrics;
+
+module.exports = (test, filter, user) => {
+
+	const uuid = user.uuid;
+	const daysBeforeTrialEnds = filter.value;
```
Comment (commuterjoy) [inline server/filters/lapsingtrialists.js]: can we not call the filter `daysLeftOnTrial` ?

File: server/filters/lapsingtrialists.js (hunk 1 above)
Comment (GlynnPhillips) [inline server/filters/lapsingtrialists.js]: Good point, adding now

File: test/filters/laspingtrialist.test.js [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+		});
+	});
+	it('Should not segment a user who has more than 7 days left on their trial', function (done) {
+		let tenDaysInFuture = new Date();
+		tenDaysInFuture = tenDaysInFuture.setDate(tenDaysInFuture.getDate() + 10);
+
+		this.mitm.on('request', function (req, res) {
+			res.statusCode = 200;
+			res.end('{ \
+				"b2b": { \
```
Comment (commuterjoy) [inline test/filters/laspingtrialist.test.js]: for clarity I usually just mock the minimum JSON structure of the test.

**Comment by commuterjoy** [comment]
+1 aside from comments

**Pull request merged**